inflight.go

```diff
@@ -33,6 +33,15 @@ func makeCall[T, V any](ctx context.Context, c *Client[T], key string, fn FetchF
 	}()
 
 	response, err := fn(ctx)
+	valueAsT, valueIsAssignableToT := any(response).(T)
+	if valueIsAssignableToT {
+		call.val = valueAsT
+	}
+	if !valueIsAssignableToT {
+		call.err = ErrInvalidType
+		return
+	}
```

Comment:
I took a look at this PR briefly and there is something I don't get.

Unless I'm wrong, the current code is something like this

```go
	if ok {
		// do something
	}
	if !ok {
		// do something else
		return
	}
```

I would suggest to refactor slightly to make things simpler

```suggestion
 	response, err := fn(ctx)
	valueAsT, valueIsAssignableToT := any(response).(T)
	if !valueIsAssignableToT {
		call.err = ErrInvalidType
		return
	}
	call.val = valueAsT
```

Reply:
I agree that's much better, thanks!

Reply:
Ah yeah, I should have spotted this. Just doing the `if ! valueIsAssignableToT` check first and exiting early is much nicer. Thanks!

Reply:
Ah yeah, I should have spotted this 🙈 Just doing the `if ! valueIsAssignableToT` check first and exiting early is much nicer. Thanks!